packages/headless/src/app/case-assist-engine/case-assist-engine.ts

```diff
@@ -0,0 +1,118 @@
+import {StateFromReducersMapObject} from '@reduxjs/toolkit';
+import {
+  buildEngine,
+  CoreEngine,
+  EngineOptions,
+  ExternalEngineOptions,
+} from '../engine';
+import {CaseAssistAppState} from '../../state/case-assist-app-state';
+import {debug, caseAssist} from '../reducers';
+import {
+  CaseAssistEngineConfiguration,
+  caseAssistEngineConfigurationSchema,
+} from './case-assist-engine-configuration';
+import {buildLogger} from '../logger';
+import {Logger} from 'pino';
+import {buildThunkExtraArguments} from '../thunk-extra-arguments';
+import {NoopPreprocessRequest} from '../../api/preprocess-request';
+import {isNullOrUndefined} from '@coveo/bueno';
+import {CaseAssistAPIClient} from '../../api/service/case-assist/case-assist-api-client';
+import {CaseAssistThunkExtraArguments} from '../case-assist-thunk-extra-arguments';
+import {setCaseAssistId} from '../../features/case-assist/case-assist-actions';
+
+export {CaseAssistEngineConfiguration};
+
+const caseassistEngineReducers = {
+  debug,
+  caseAssist,
+};
+type CaseAssistEngineReducers = typeof caseassistEngineReducers;
+type CaseAssistEngineState =
+  StateFromReducersMapObject<CaseAssistEngineReducers> &
+    Partial<CaseAssistAppState>;
+
+/**
+ * The engine for powering case assist experiences.
+ */
+export interface CaseAssistEngine<State extends object = {}>
+  extends CoreEngine<
+    State & CaseAssistEngineState,
+    CaseAssistThunkExtraArguments
+  > {}
```

Comment:
Will the `CaseAssistEngine` eventually expose some kind of helper function similar to the search one ( eg: `executeFirstSearch` ) ?

Or still to be determined ?

Reply:
It will log a `case start` analytics event but I don't think that'll be optional. But I guess it something like that could exist in the future.